aiida_quantumespresso/parsers/base.py

```diff
@@ -3,15 +3,128 @@
 
 All `Parser` implementations in `aiida-quantumespresso` must use this base class, not `aiida.parsers.Parser`.
 """
-from aiida.parsers import Parser as _BaseParser
+import re
+import typing
+
+from aiida.parsers import Parser
+from aiida.common import AttributeDict
+from aiida.engine import ExitCode
+from aiida_quantumespresso.utils.mapping import get_logging_container
+from aiida_quantumespresso.parsers.parse_raw.base import convert_qe_time_to_sec
+
+__all__ = ('BaseParser',)
+
+
+class BaseParser(Parser):  # pylint: disable=abstract-method
+    """Custom ``Parser`` class for ``aiida-quantumespresso`` parser implementations."""
+
+    class_error_map = {}
+    class_warning_map = {}
+
+    base_error_map = {
+        'Maximum CPU time exceeded': 'ERROR_OUT_OF_WALLTIME',
+    }
+    base_warning_map = {
+        'Warning:': None,
+        'DEPRECATED:': None,
+    }
+
+    @classmethod
+    def get_error_map(cls):
+        """The full error map of the parser class."""
+        error_map = cls.base_error_map.copy()
+        error_map.update(cls.class_error_map)
+        return error_map
+
+    @classmethod
+    def get_warning_map(cls):
+        """The full error map of the parser class."""
+        warning_map = cls.base_warning_map.copy()
+        warning_map.update(cls.class_warning_map)
+        return warning_map
+
+    def _retrieve_parse_stdout(self, **kwargs) -> typing.Tuple[str, dict, AttributeDict]:
+        """Retrieve and parse the ``stdout`` content of a Quantum ESPRESSO calculation.
+
+        :returns: size 3 tuple with the stdout content, parsed data and log messages
+        """
+        logs = get_logging_container()
+
+        filename_stdout = self.node.get_option('output_filename')
+
+        if filename_stdout not in self.retrieved.list_object_names():
+            logs.error.append('ERROR_OUTPUT_STDOUT_MISSING')
+            return {}, logs
+
+        try:
+            with self.retrieved.open(filename_stdout, 'r') as handle:
+                stdout = handle.read()
+        except OSError:
+            logs.error.append('ERROR_OUTPUT_STDOUT_READ')
+            return {}, logs
+
+        parsed_data, stdout_logs = self.parse_stdout(stdout, **kwargs)
+
+        for log_type, log_items in stdout_logs.items():
```

Comment:
`log_type` here would be like `warning` or `error` etc.? If so, probably `log_level` is better because that is the canonical term for it.